src/osuTK/Platform/Windows/WinRawJoystick.cs

```diff
@@ -336,8 +336,11 @@ public unsafe bool ProcessEvent(IntPtr raw)
                         Array.Resize(ref DataBuffer, report_count);
                     }
 
-                    UpdateAxes(rin, stick);
-                    UpdateButtons(rin, stick);
+                    lock (stick)
```

Comment:
What's this? It's guaranteed to be an exclusive lock as far as I can tell?

Reply:
prevents reading of the JoystickState while it's being populated. not having the reading and creation of JoystickState was causing ghost release & press events while a button was being held

Reply:
So this compliments the lock in `GetState()`? Is there a reason this couldn't be `lock (UpdateLock)`?

Reply:
the lock needs to be tied to the specific Device and that means either locking on the Device itself or having a lock object defined under Device. since only WinRawJoystick needs this workaround I opted to not add a new member to Device.